### PR TITLE
Feature events calculator

### PR DESCRIPTION
**Pull request by AlejandroNo4**
For this milestone, I connected the files with event handlers. I applied code to handle the states of the main object.

I made a significant change in the calculate.js file. Since in the previous milestone, I couldn't tell if it was working.

Since the previous milestone was somewhat confusing, I am still not 100% sure what the required behavior of the calculator is.

It is a simple calculator, and it has these characteristics:

- When the total is 0, the only buttons available are the numbers.

- **The calculator only accepts binary operations.**

- When a total has been given, the only button available will be "AC" to return to 0.

As I mentioned previously, it was a bit confusing getting to this point. Therefore, if changes are required, please make them a bit more explicit.

Thank you in advance!

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,14 +1,29 @@
-import React from 'react';
-import Display from './Display';
+import React, { useState } from 'react';
 import ButtonPanel from './ButtonPanel';
 import calculate from '../logic/calculate';
+import Display from './Display';
+
+const App = () => {
```
Comment (rindrajosia) [inline src/components/App.js]: - [x] [REQUIRED] it is converted to a stateful component using class ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/stateful.png))
- [x] [REQUIRED] it has state initialized in constructor ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/stateful.png))

File: src/components/App.js [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+  const [total, updateTotal] = useState('0');
+  const [next, updateNext] = useState('0');
+  const [operation, updateOperation] = useState('');
+
+  const handleClick = (btnName) => {
```
Comment (rindrajosia) [inline src/components/App.js]: - [x] [REQUIRED] `handleClick` should be implemented in constructor

File: src/components/App.js [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+  };
 
-export default function App() {
-  calculate();
   return (
     <>
-      <Display result="0" />
-      <ButtonPanel />
+      <div>
+        <Display result={next.toString()} />
```
Comment (rindrajosia) [inline src/components/App.js]: - [x] [REQUIRED] it passes the `total` or `next` value from state to `Display` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passToatalToDisplay.png))

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [ ] [REQUIRED] Your APP crashed when I divide a number by 0. Instead, your APP should display an error message like "Impossible".


- [ ] [REQUIRED] I cannot make an operation directly with the previous result.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- to update the live demo link as it is broken

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by AlejandroNo4** [comment]
I added the required changes, but there is an important point to highlight.❗️❗️❗️

Requirements establish the initial values ​​of the total, next, and operation must be set null by default. This was causing me a conflict with PropTypes as it requires props to be a string in Display.

The closest I could think of was to give it a value of '-' so the calculator now has an "on" behavior since it is necessary to click on "AC" to start using it.

**Review by voscarmv — APPROVED** [review]
# Project status: Approved 💯 ✨ 😎 👍 
### You may merge this branch and move forward ⛙🛣️ 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻
# Happy Coding! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**